huaweicloud/services/workspace/resource_huaweicloud_workspace_app_rule.go

```diff
@@ -0,0 +1,474 @@
+	identifyCondition, ok := productRule["identify_condition"].(string)
+	if !ok || identifyCondition == "" {
+		return errors.New("identify_condition is required in product_rule")
+	}
+
+	publisher, _ := productRule["publisher"].(string)
+	productName, _ := productRule["product_name"].(string)
+	processName, _ := productRule["process_name"].(string)
+	starOrEmptyCount := 0
+	if publisher == "*" || publisher == "" {
+		starOrEmptyCount++
+	}
+	if productName == "*" || productName == "" {
+		starOrEmptyCount++
+	}
+	if processName == "*" || processName == "" {
+		starOrEmptyCount++
+	}
+	if starOrEmptyCount == 3 {
+		return errors.New("publisher, product_name, and process_name cannot be all '*' or empty in product_rule")
+	}
+
+	return nil
+}
+
+func buildAppPathRule(pathRules []interface{}) map[string]interface{} {
+	if len(pathRules) < 1 {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"path": utils.PathSearch("path", pathRules[0], nil),
+	}
+}
+
+func buildAppProductRule(productRules []interface{}) map[string]interface{} {
+	if len(productRules) < 1 {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"identify_condition": utils.PathSearch("identify_condition", productRules[0], nil),
+		"publisher":          utils.PathSearch("publisher", productRules[0], nil),
+		"product_name":       utils.PathSearch("product_name", productRules[0], nil),
+		"process_name":       utils.PathSearch("process_name", productRules[0], nil),
+		"support_os":         utils.PathSearch("support_os", productRules[0], "Windows"),
+		"version":            utils.ValueIgnoreEmpty(utils.PathSearch("version", productRules[0], nil)),
+		"product_version":    utils.ValueIgnoreEmpty(utils.PathSearch("product_version", productRules[0], nil)),
+	}
+}
+
+func buildAppRuleConfig(rules []interface{}) map[string]interface{} {
+	if len(rules) < 1 {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"scope":        utils.PathSearch("scope", rules[0], nil),
+		"product_rule": buildAppProductRule(utils.PathSearch("product_rule", rules[0], make([]interface{}, 0)).([]interface{})),
+		"path_rule":    buildAppPathRule(utils.PathSearch("path_rule", rules[0], make([]interface{}, 0)).([]interface{})),
+	}
+}
+
+func buildCreateAppRuleBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name":        d.Get("name"),
+		"rule":        buildAppRuleConfig(d.Get("rule").([]interface{})),
+		"description": d.Get("description"),
+	}
+}
+
+func resourceAppRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.NewServiceClient("workspace", cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating Workspace client: %s", err)
+	}
+
+	// Validate product_rule constraints before creation
+	err = validateProductRule(d)
+	if err != nil {
+		return diag.Errorf("invalid product_rule: %s", err)
+	}
```

Comment:
There is unnecessary to validate the rule configuration, if the request body is incorrect, the service will report a corresponding error.